src/main/kotlin/org/jitsi/nlj/rtp/RtxPacket.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.nlj.rtp
+
+import org.jitsi.nlj.util.shiftPayloadLeft
+import org.jitsi.nlj.util.shiftPayloadRight
+import org.jitsi.rtp.extensions.bytearray.putShort
+import org.jitsi.rtp.rtp.RtpPacket
+import org.jitsi.rtp.util.getShortAsInt
+
+/**
+ * https://tools.ietf.org/html/rfc4588#section-4
+ *
+ *  0                   1                   2                   3
+ *  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                         RTP Header                            |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |            OSN                |                               |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               |
+ * |                  Original RTP Packet Payload                  |
+ * |                                                               |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ */
+class RtxPacket {
+    companion object {
+        fun getOriginalSequenceNumber(rtxPacket: RtpPacket): Int =
+            rtxPacket.buffer.getShortAsInt(rtxPacket.offset + rtxPacket.headerLength)
```

Comment:
This doesn't check the length. Are we okay with the pipeline throwing exceptions on certain packets?

Reply:
yeah i think we'll need a pass for hardening with regards to length and stuff...i kinda like throwing for now as in our testing anytime it happens is a bug.  plus, arguably we should be able to handle an exception for one packet in the pipeline and not have it screw things up completely.  but we probably do want a good scheme for protecting against malicious packet fields